### DISCUSSION TITLE
How to Become a software engineer

### DISCUSSION THREAD

**Discussion by danukaji-M** [community/community#131743] [Programming Help] [answered]
### Select Topic Area

Software Engineering 

### Body

i need to know about how to become a software engineer and, if i learn about Discrete Mathematics , Linear Algebra ,Probability and Statistics ,Calculus , Number Theory, Algebra, Mathematical Logic, Information Theory what can i do with them and i need to learn about them because i love mathematics 
what can i do with learning them, i like to learn about ai and ml, is that mathematical knowledge enough for it

**Comment by LiteBrite82** [comment]
Thanks for posting in the GitHub Community, @danukaji-M ! 

We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.

**Comment by ghost** [comment, marked as answer]
Hi, this knowledge is the basis for work in the field of artificial intelligence and machine learning. Knowledge of mathematics can help you understand the basic algorithms and models that underlie artificial intelligence, and can also help you develop and optimize machine learning algorithms. In addition, you will better understand the mathematical models and methods used in AI and ML.

**Reply by danukaji-M** [reply]
Thanks @komoto-0
